### PR TITLE
Doc style and link enhancements

### PR DESCRIPTION
**Pull request by michelle-purcell**
This PR fixes style issues (title capitalization, one sentence per line...), links, and some grammar as per the preferred Quarkus Contributor guidelines for writing tutorials.
This also harmonizes the content with product docs downstream for the 3.2 release.

### CODE REVIEW

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc
Comment (sheilamjones) [inline docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc:87]: Minor/non-urgent suggestion: Should the capitalization of the URL text be in sentence case.

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc
Comment (sheilamjones) [inline docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc:119]: Minor/non-urgent: Capitalization of the URL text. Should this be "Security identity customization"

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc
Comment (sheilamjones) [inline docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc:169]: Suggestion: For clarity, propose to say "If no matching `JWK` key is available...."

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc
Comment (sheilamjones) [inline docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc:172]: Same comment as for line 169, for clarity, should we say "...`JWK` key..." here?

Comment (michelle-purcell) [reply docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc]: @sheilamjones - +1. Thanks for flagging here and elsewhere. All instances are now fixed.

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc
Comment (michelle-purcell) [reply docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc]: @sheilamjones - Fixed. Thanks!

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc
Comment (michelle-purcell) [reply docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc]: @sheilamjones  Fixed. Good catch. Thanks :100: 
_Your comment also triggered a review of the capitalization in some level 3 section headings. Fixed a few of those too._

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc:1046]:
```suggestion
=== Check errors in the log
```

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc:429]: I'm not sure the label name change is necessary, there might already be be external links pointing to it (from the code flow docs for example, from somewhere else), other labels follow a different pattern, it all can be realigned but it is not worth it IMHO :-)

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc:426]: Sure, may be just underneath `Testing section` above - but I can't promise I'll remember this comment :-), may be worth opening an issue ? Or I can push an update to your PR ?

**Comment by sberyozkin** [comment]
Hi @michelle-purcell It all looks very good, significant improvements to the quality of the text, thanks. Thanks @sheilamjones too.

I've only opened a couple of suggestions - if you are OK with them then I can push the updates to your PR during the next day or two and merge it

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc
Comment (sberyozkin) [reply docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc:1046]: Otherwise it reads as if we are checking `log errors` which may not be necessary related to OIDC. I have to admit I'm not sure what reads better `in the log` or `in the logs`...

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc
Comment (michelle-purcell) [reply docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc:426]: @sberyozkin - :+1: +1 - An update to my PR would be perfect. Thank you.

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc
Comment (michelle-purcell) [reply docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc]: @sberyozkin - Makes sense. Thanks!

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc
Comment (michelle-purcell) [inline docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc:1046]:
```suggestion
=== Check errors in the logs
```

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc
Comment (michelle-purcell) [reply docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc:429]: @sberyozkin - Good point. Fixed. I think I was trying to fix a broken anchor, and somehow I decided to recreate it this way :thinking:  I have reverted it back and kept the anchor fix in but with the original ID. Thanks!

**Comment by michelle-purcell** [comment]
@sberyozkin - Thanks so much for reviewing the PR before I head off for PTO :tada:  I have fixed all suggestions and rebased them. There's just that one outstanding intro bit left ^^ that you are going to add and merge into the PR, if git lets you. :green_heart: 

@sheilamjones - Thank you too, Sheila for reviewing :green_heart:

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc
Comment (sberyozkin) [reply docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc:428]: Sounds good @michelle-purcell

**Review by sberyozkin — APPROVED** [review]

**Comment by sberyozkin** [comment]
Looks like it was cancelled and I missed merging it.

**Comment by sberyozkin** [comment]
It is upstream only though so let me merge it now, though I'll rebase

**Comment by sberyozkin** [comment]
Hmm, `exec-maven-plugin:3.0.0:exec (assemble-downstream-doc) @ quarkus-documentation` is hanging, but I think it just finished on my laptop, I was checking Michelle's branch.

**Comment by sberyozkin** [comment]
@gsmet I'm not sure why this PR is hanging, I don't see anything in the updated doc that could've caused it. Please review and merge if it looks fine. on PTO tomorrow

**Comment by inoxx03** [comment]
@sberyozkin Thasnk for the head's up. TBH, I have not found the cause yet.  trheI'm not sure but `assemble-downstream-doc` might be one of the post-processing jobs that we use to transform parts of thew upstream doc's AsciiDoc source to get around the downstream publishing system limitations.

 I'll come back and investigate further today, just have to finish some release-related tasks.

**Comment by sheilamjones** [comment]
Apologies for slow reply @sberyozkin. Was tied up with 3.2 release tasks yesterday. Yes, as Stefan suggested, I think the assemble-downstream.doc topic relates to the downstreaming script with the post-processing fixes to enable us to bring the content downstream. Shall investigate further.

**Comment by sberyozkin** [comment]
@sheilamjones No problems at all Sheila, thanks for the update.

Unfortunately it is still failing, something in this PR or in some other PRs is causing this task to fail.

@rsvoboda Rostislav, can you spot it ? Your fix is in Michelle's branch: https://github.com/michelle-purcell/quarkus/commits/security-doc-fixes-3-2-bearer-concept

File: docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc
Comment (rsvoboda) [inline docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc:438]: This noise makes the build of downstream docs to be stuck

there should be no `+` and `====` on solo lines

CC @sberyozkin @michelle-purcell

Comment (sberyozkin) [reply docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc]: @rsvoboda @michelle-purcell It took me may be 5 rounds to get past the downstream task, I've ended removing all of standalone `+` and `====` and it looked like it fixed. Michelle, not sure if te end result would be what you expected it to be without these characters, so I guess you should have a look before me merging this PR

Thanks

**Comment by sberyozkin** [comment]
Michelle, I'm not sure the numbering scheme works, there are a few places with `1.` where another `.1` follows in the same section after one paragraph etc. Can we consider removing it ?

**Comment by michelle-purcell** [comment]
Thanks @sberyozkin. Apologies for not getting to this sooner. Between PTO and other priorities... 
Some of the numbering markup edits were introduced to fix bad numbering/style on the downstream publishing output too and to follow the advice in the upstream tutorial/how-to templates. 
I will test and investigate today and test the output of the markup on both publishing platforms. Thanks for flagging :-)

**Comment by sberyozkin** [comment]
Hi Michelle @michelle-purcell , missed your comment, sorry. Sure please check it

**Comment by michelle-purcell** [comment]
Closing this PR as it has languished and overlaps with other changes. @rolfedh is taking over the edits for this guide in a fresh PR. Thanks @rolfedh. Please see the comments above for your reference. Thanks.

**Pull request closed without merging**
